app/adapters/signal_adapter/outbound.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module SignalAdapter
+  class Outbound < ApplicationJob
+    queue_as :default
+    def self.send!(message)
+      recipient = message.recipient
+      return unless recipient&.phone_number
+
+      perform_later(text: message.text, recipient: recipient)
+    end
+
+    def self.send_welcome_message!(contributor)
+      return unless contributor&.phone_number
+
+      welcome_message = [Setting.onboarding_success_heading, Setting.onboarding_success_text].join("\n")
+      perform_later(text: welcome_message, recipient: contributor)
+    end
+
+    def perform(text:, recipient:)
+      url = URI.parse("#{Setting.signal_rest_cli_endpoint}/v2/send")
+      header = {
+        Accept: 'application/json',
+        'Content-Type': 'application/json'
+      }
+      data = {
+        number: Setting.signal_phone_number,
+        recipients: [recipient.phone_number],
+        message: text
+      }
+      req = Net::HTTP::Post.new(url.to_s, header)
+      req.body = data.to_json
+      res = Net::HTTP.start(url.host, url.port) do |http|
+        http.request(req)
+      end
+      res.value # may raise exception
```

Comment:
When and why does this raise an exception?

Reply:
I think when the service is not reachable, it will respond with a non-JSON HTTP response and then parsing it will fail.

Reply:
Well, if the server is not reachable, it will not respond at all. However, if Signal is not reachable, it will respond with a non-JSON response or, I guess, if we send a bad request.